rotkehlchen/db/history_events.py

```diff
@@ -642,36 +664,28 @@ def get_history_events_count(
         the number of events if any limit is applied, otherwise the second value matches
         the first.
         """
-        prepared_query, bindings = query_filter.prepare(with_pagination=False)
-        # we need to select everything because any column could be used in the filter
-        query = f'SELECT {query_filter.get_columns()} {query_filter.get_join_query()} {prepared_query}'  # noqa: E501
-        if group_by_event_ids:
-            query = f'SELECT event_identifier FROM ({query}) GROUP BY event_identifier'
-        query = f'SELECT COUNT(*) FROM ({query})'
-        count_without_limit = cursor.execute(query, bindings).fetchone()[0]
-
-        if entries_limit is not None:
-            prepared_query, bindings = query_filter.prepare(
-                with_group_by=False,
-                special_free_query=True,
-                with_order=False,
-                with_pagination=False,
-            )
+        (
+            (premium_query, premium_bindings),
+            (free_query, free_bindings),
+        ) = self._create_history_events_query(
+            filter_query=query_filter,
+            group_by_event_ids=group_by_event_ids,
+            entries_limit=FREE_HISTORY_EVENTS_LIMIT if entries_limit is None else entries_limit,
+        )
+
+        count_without_limit = cursor.execute(
+            f'SELECT COUNT(*) FROM ({premium_query})',
+            premium_bindings,
+        ).fetchone()[0]
+
+        if entries_limit is None:
+            return count_without_limit, count_without_limit
```

Comment:
after this check is when you can call `_create_history_events_query` with `has_premium=False`